### PR TITLE
Enhancement: Add composer scripts to ease development

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] adds composer scripts to ease development

Run

```
$ composer cs
```

to fix and check coding style issues.

Run

```
$ composer test
```

to run the tests.

Run

```
$ composer all
```

if you are extra lazy to run both.

### CODE REVIEW

**Comment by simonengelhardt** [comment]
Thanks for another nice PR @localheinz. We're a bit hesitant to go down the `make` route, since we'd like to keep tool dependencies to a minimum (considering support for people on Windows or Mac OS without Xcode tools).

How about using [Composer custom commands](https://getcomposer.org/doc/articles/scripts.md#writing-custom-commands) instead (adding a `test` and a `cs` command)? I realize that wouldn't let you with no arguments (combining test and cs), but I think that would be an ok trade-off.

**Comment by localheinz** [comment]
@simonengelhardt 

👍 Updated!

File: composer.json [hunk 1]
```diff
@@ -28,5 +28,17 @@
         "psr-4": {
             "Facebook\\InstantArticles\\": "tests/Facebook/InstantArticles/"
         }
+    },
+    "scripts": {
+        "all": [
+            "composer cs",
```
Comment (simonengelhardt) [inline composer.json]: We can simply use `@cs` here (see https://getcomposer.org/doc/articles/scripts.md#referencing-scripts)

File: composer.json [hunk 2]
```diff
@@ -28,5 +28,17 @@
...
+            "composer test"
```
Comment (simonengelhardt) [inline composer.json]: ^ `@test`

File: composer.json [hunk 3]
```diff
@@ -28,5 +28,17 @@
...
+        ],
+        "cs": [
+            "vendor/bin/phpcbf --standard=phpcs.xml -p || vendor/bin/phpcs --standard=phpcs.xml -p"
```
Comment (simonengelhardt) [inline composer.json]: We can take advantage of the PATH magic mentioned [here](https://getcomposer.org/doc/articles/scripts.md#writing-custom-commands) and simply use:

```
phpcbf --standard=phpcs.xml -p || phpcs --standard=phpcs.xml -p
```

File: composer.json [hunk 4]
```diff
@@ -28,5 +28,17 @@
...
+        ],
+        "test": [
+            "vendor/bin/phpunit"
```
Comment (simonengelhardt) [inline composer.json]: ^ `phpunit`

File: composer.json [hunk 5]
```diff
@@ -28,5 +28,17 @@
...
+    },
+    "scripts": {
+        "all": [
+            "composer cs",
+            "composer test"
+        ],
+        "cs": [
+            "vendor/bin/phpcbf --standard=phpcs.xml -p || vendor/bin/phpcs --standard=phpcs.xml -p"
+        ],
+        "test": [
```
Comment (simonengelhardt) [inline composer.json]: We could "depend" on install by adding `"@composer install",` as in the example [here](https://getcomposer.org/doc/articles/scripts.md#calling-composer-commands)

**Comment by localheinz** [comment]
@simonengelhardt 

Thank you for the heads-up, wasn't aware of it!

One thing, though, adjusting `composer.json` to

```json
{
    "scripts": {
        "all": [
            "@cs",
            "@test"
        ],
        "cs": [
            "@composer install",
            "phpcbf --standard=phpcs.xml -p || phpcs --standard=phpcs.xml -p"
        ],
        "test": [
            "@composer install",
            "phpunit"
        ]
    }
}
```

and then running

```
$ rm -rf vendor && composer test
```

or 

```
$ rm -rf vendor && composer test
```

fails, because `composer install` is never executed.

Not sure why, though, composer is up-to-date:

```
$ composer --version
Composer version 1.0.3 2016-04-29 16:30:15
```

**Comment by simonengelhardt** [comment]
@localheinz

Thanks for implementing all my small comments 👍

I just verified that the `@composer` syntax indeed seems to be broken. Let's assume people install Composer globally (which is what we instruct people to do in the README) and have `composer` available as an executable command for now so we can merge in what you have come up with here, file an issue on the Composer repo to fix, and then we can update to use `@composer` afterwards.

**Comment by localheinz** [comment]
Thank you, @simonengelhardt!

**Pull request merged**